api/src/main/java/org/openmrs/module/referencedemodata/patient/DemoPatientGenerator.java

```diff
@@ -407,6 +439,79 @@ private void createDemoLabObs(Patient patient, Encounter encounter, Location loc
+        Date endDateTime = toDate(toLocalDateTime(visitTime).plusMinutes(randomBetween(60, 90)));
+        appointment.setStartDateTime(startDateTime);
+        appointment.setEndDateTime(endDateTime);
+        appointment.setAppointmentKind(AppointmentKind.Scheduled);
+        
+        appointment.setService(getAppointmentServiceDefinitionService().getAllAppointmentServices(false).get(randomBetween(0, 20) % 2));
+        appointment.setAppointmentAudits(new HashSet<>());
+        Set<AppointmentProvider> appointmentProviders = new HashSet<>();
+        AppointmentProvider appointmentProvider = new AppointmentProvider();
+        appointmentProvider.setAppointment(appointment);
+        appointmentProvider.setProvider(getProviderService().getProviderByIdentifier("admin"));
+        appointmentProvider.setResponse(AppointmentProviderResponse.ACCEPTED);
+
+        appointmentProviders.add(appointmentProvider);
+        appointment.setProviders(appointmentProviders);
+        Context.addProxyPrivilege(PrivilegeConstants.MANAGE_APPOINTMENTS);
+        getAppointmentsService().validateAndSave(appointment);
+        Context.removeProxyPrivilege(PrivilegeConstants.MANAGE_APPOINTMENTS);
+        
+	}
+	
+	private void createDemoPatientProgram(Patient patient, Program program, Date visitTime) {
+		
+		Date startDate = toDate(toLocalDateTime(visitTime).plusHours((randomBetween(0, 72))));
+		ProgramWorkflowState state1 = program.getAllWorkflows().stream().findFirst().get().getStates().stream().findFirst().get();
+
+		PatientProgram patientprogram = new PatientProgram();
+		patientprogram.setProgram(program);
+		patientprogram.setPatient(patient);
+		patientprogram.setDateEnrolled(startDate);
+		patientprogram.setDateCompleted(null);
+
+		PatientState patientstate1 = new PatientState();
+		patientstate1.setStartDate(startDate);
+		patientstate1.setState(state1);
+		
+		patientprogram.getStates().add(patientstate1);
+		patientstate1.setPatientProgram(patientprogram);
+		
+		getProgramWorkflowService().savePatientProgram(patientprogram); 
+	}
+	
+	private Encounter createDemoOrdersEncounter(Patient patient, Date encounterTime, Location location) {
+		Encounter encounter = createEncounter("Consultation", patient, encounterTime, location);
+		return encounter;
+	}
+	
+	private void createDemoOrder(Encounter encounter, OrderType orderType, List<Concept> orderConcepts) {
+		OrderService orderService = getOrderService();
+		List<CareSetting> careSettings = orderService.getCareSettings(false);
+		if ("Test Order".equalsIgnoreCase(orderType.getName())) {
+			TestOrder order = new TestOrder();
+			order.setAction(Order.Action.NEW);
+			order.setUrgency(Order.Urgency.ROUTINE);
+			order.setPatient(encounter.getPatient());
+			order.setOrderType(orderType);
+			order.setConcept(orderConcepts.get(randomBetween(0, orderConcepts.size())));
+			order.setOrderer(getProviderService().getProviderByIdentifier("admin"));
```

Comment:
We actually have multiple providers in the demo set. It would be best if the default was the `doctor` user rather than `admin`.